Cubical/Categories/Functor/ComposeProperty.agda

```diff
@@ -0,0 +1,420 @@
+          ∙ cong (λ x → x ⋆⟨ D ⟩ _ ⋆⟨ D ⟩ _) (sym (D .⋆IdR _))
+          ∙ cong (λ x → _ ⋆⟨ D ⟩ x ⋆⟨ D ⟩ _ ⋆⟨ D ⟩ _) (sym (h .snd .sec))
+          ∙ cong (λ x → x ⋆⟨ D ⟩ _ ⋆⟨ D ⟩ _) (sym (D .⋆Assoc _ _ _))
+          ∙ cong (λ x → x ⋆⟨ D ⟩ _) (D .⋆Assoc _ _ _)
+          ∙ D .⋆Assoc _ _ _
+          ∙ (λ i → ⋆InvRMove h m-eq (~ i) ⋆⟨ D ⟩ ⋆InvRMove h' p (~ i))
+          ∙ sym (F .F-seq _ _)
+
+        m-path : l₀ ⋆⟨ C ⟩ m' .fst ≡ m .fst ⋆⟨ C ⟩ l
+        m-path = isFullyFaithful→Faithful {F = F} (w-equiv .fullfaith) _ _ _ _ Fm-path
+
+        Gm-path : G .F-hom l₀ ⋆⟨ E ⟩ G .F-hom (m' .fst) ≡ G .F-hom (m .fst) ⋆⟨ E ⟩ G .F-hom l
+        Gm-path = sym (G .F-seq _ _) ∙ cong (G .F-hom) m-path ∙ G .F-seq _ _
+
+        Gm-path' : G .F-hom l ⋆⟨ E ⟩ G .F-hom (m' .snd .inv) ≡ G .F-hom (m .snd .inv) ⋆⟨ E ⟩ G .F-hom l₀
+        Gm-path' = ⋆InvLMove (F-Iso {F = G} m) (sym (⋆InvRMove (F-Iso {F = G} m') Gm-path ∙ E .⋆Assoc _ _ _))
+
+      isContrMor' .snd (g₁ , coh₁) i .fst =
+         (⋆InvLMove (k h₀) (sym (isContrMor' .fst .snd c₀ c₁ h₀ h₁ l₀ p₀))
+        ∙ sym (⋆InvLMove (k h₀) (sym (coh₁ c₀ c₁ h₀ h₁ l₀ p₀)))) i
+      isContrMor' .snd x@(g₁ , coh₁) i .snd =
+        isProp→PathP (λ i → isPropΠ6 (λ c c' h h' l _ →
+          E .isSetHom
+            (G .F-hom l ⋆⟨ E ⟩ k h' .fst)
+            (k h .fst ⋆⟨ E ⟩ isContrMor' .snd x i .fst)))
+        (isContrMor' .fst .snd) coh₁ i
+
+    module _ {c c'} {d d'}
+      (f : D [ d , d' ])(h : CatIso D (F .F-ob c) d)(h' : CatIso D (F .F-ob c') d')
+      where
+
+      liftL : C [ c , c' ]
+      liftL = invEq (_ , w-equiv .fullfaith _ _) (h .fst ⋆⟨ D ⟩ f ⋆⟨ D ⟩ h' .snd .inv)
+
+      liftL-eq : F .F-hom liftL ⋆⟨ D ⟩ h' .fst ≡ h .fst ⋆⟨ D ⟩ f
+      liftL-eq =
+        sym (⋆InvRMove (invIso h')
+          (sym (secEq (_ , w-equiv .fullfaith _ _) (h .fst ⋆⟨ D ⟩ f ⋆⟨ D ⟩ h' .snd .inv))))
+
+    isContrMor : (d d' : D .ob)(f : D [ d , d' ]) → isContr (Mor d d' f)
+    isContrMor d d' f = Prop.rec2 isPropIsContr
+      (λ (c₀ , h₀) (c₁ , h₁) →
+        isContrMor' d d' f c₀ h₀ c₁ h₁ (liftL f h₀ h₁) (liftL-eq f h₀ h₁))
+      (w-equiv .esssurj d) (w-equiv .esssurj d')
+
+    Ext-hom : {d d' : D .ob}(f : D [ d , d' ]) → E [ Ext-ob d , Ext-ob d' ]
+    Ext-hom f = isContrMor _ _ f .fst .fst
+
+    liftL⋆ : ∀ {c c' c''} {d d' d''}
+      (f : D [ d , d' ])(g : D [ d' , d'' ])
+      (h : CatIso D (F .F-ob c) d)(h' : CatIso D (F .F-ob c') d')(h'' : CatIso D (F .F-ob c'') d'')
+      → liftL (f ⋆⟨ D ⟩ g) h h'' ≡ liftL f h h' ⋆⟨ C ⟩ liftL g h' h''
+    liftL⋆ f g h h' h'' = isFullyFaithful→Faithful {F = F} (w-equiv .fullfaith) _ _ _ _
+      (⋆CancelR h'' path ∙ sym (F .F-seq _ _))
+      where
+      path : _
+      path = liftL-eq (f ⋆⟨ D ⟩ g) h h''
+        ∙ sym (D .⋆Assoc _ _ _)
+        ∙ cong (λ x → x ⋆⟨ D ⟩ _) (sym (liftL-eq f h h'))
+        ∙ D .⋆Assoc _ _ _
+        ∙ cong (λ x → F .F-hom (liftL f h h') ⋆⟨ D ⟩ x) (sym (liftL-eq g h' h''))
+        ∙ sym (D .⋆Assoc _ _ _)
+
+    Ext : Functor D E
```

Comment:
I'm a little lost at this point, is this a local definition or at the toplevel of the module?

Reply:
(If it's at the toplevel then `Ext` is not a very good name unless you're doing homological algebra :-) )

Reply:
It's local and only used in this proof, and `Ext` stands for "the functor Extended from the given one" :P

Reply:
(I guess I don't need to rename it?)

Reply:
Yeah, it's fine :-)